assets/app/scripts/controllers/overview.js

```diff
@@ -58,6 +71,99 @@ angular.module('openshiftConsole')
       console.log("podsByServiceByLabel", $scope.podsByServiceByLabel);      
     };
 
+    var podRelationships = function() {
```

Comment:
building all the pod maps in one function (podsByDeployment, podsByService, monopodsByService) would let us avoid building and evaluating the same label selectors on the same pods repeatedly